packages/core/src/plugins/rsdoctor.ts

```diff
@@ -45,9 +48,14 @@ export const pluginRsdoctor = (): RsbuildPlugin => ({
 
       let isAutoRegister = false;
 
+      function isRsdoctorPlugin(plugin: Configuration['plugins'] & { isRsdoctorPlugin?: boolean }) {
+        return plugin != null && typeof plugin.isRsdoctorPlugin === 'boolean';
+      }
+
       for (const config of bundlerConfigs) {
-        const registered = config.plugins?.some(
-          (plugin) => plugin?.constructor?.name === pluginName,
+        // If user has add Rsdoctor plugin in config file, will return.
+        const registered = config.plugins?.some(          
+          (plugin) => isRsdoctorPlugin(plugin as unknown as Configuration['plugins'] & { isRsdoctorPlugin?: boolean }) || plugin?.constructor?.name === pluginName,
```

Comment:
We can declare a type to reuse it

```suggestion
type MaybeRsdoctorPlugin = Configuration['plugins'] & { isRsdoctorPlugin?: boolean })
```